js_modules/dagit/packages/core/src/partitions/AssetJobPartitionsView.tsx

```diff
@@ -0,0 +1,139 @@
+import {Box, Button, Colors, Subheading} from '@dagster-io/ui';
+import React from 'react';
+
+import {useAssetGraphData} from '../asset-graph/useAssetGraphData';
+import {LaunchAssetExecutionButton} from '../assets/LaunchAssetExecutionButton';
+import {usePartitionHealthData} from '../assets/PartitionHealthSummary';
+import {useViewport} from '../gantt/useViewport';
+import {repoAddressToSelector} from '../workspace/repoAddressToSelector';
+import {RepoAddress} from '../workspace/types';
+
+import {JobBackfillsTable} from './JobBackfillsTable';
+import {CountBox} from './OpJobPartitionsView';
+import {PartitionState, PartitionStatus} from './PartitionStatus';
+
+export const AssetJobPartitionsView: React.FC<{
+  pipelineName: string;
+  partitionSetName: string;
+  repoAddress: RepoAddress;
+}> = ({partitionSetName, repoAddress, pipelineName}) => {
+  const {viewport, containerProps} = useViewport();
+  const repositorySelector = repoAddressToSelector(repoAddress);
+
+  const assetGraph = useAssetGraphData('', {
+    pipelineSelector: {
+      pipelineName,
+      repositoryName: repoAddress.name,
+      repositoryLocationName: repoAddress.location,
+    },
+  });
+
+  const assetHealth = usePartitionHealthData(assetGraph.graphAssetKeys);
+  const partitionNames = Array.from(new Set<string>(assetHealth.flatMap((a) => a.keys)));
+  const jobHealth = React.useMemo(
+    () =>
+      Object.fromEntries(
+        partitionNames.map((p) => [
+          p,
+          assetHealth.every((asset) =>
+            p in asset.statusByPartition ? asset.statusByPartition[p] === true : true,
+          )
+            ? PartitionState.SUCCESS
+            : PartitionState.MISSING,
+        ]),
+      ),
+    [assetHealth, partitionNames],
+  );
+
+  const [pageSize, setPageSize] = React.useState(60);
+  const [offset, setOffset] = React.useState<number>(0);
+  const [showSteps, setShowSteps] = React.useState(false);
+
+  React.useEffect(() => {
+    if (viewport.width && !showSteps) {
+      // magical numbers to approximate the size of the window, which is calculated in the step
+      // status component.  This approximation is to make sure that the window does not jump as
+      // the pageSize gets recalculated
+      const _approximatePageSize = Math.ceil((viewport.width - 330) / 32) - 3;
```

Comment:
Possible to clarify these values with some consts?

Reply:
Oh man good call on this one...